ObjC/Encoder+ObjC.mm

```diff
@@ -25,7 +25,7 @@
 
 
 @interface NSObject (FleeceInternal)
-- (void) fl_encodeTo: (Encoder*)enc;
+- (void) fl_encodeTo: (FLEncoderImpl*)enc;
```

Comment:
`FLEncoderImpl*` is literally identical to `FLEncoder` (there's a typedef at Fleece+ImplGlue.hh:44), so this makes `fl_encodeTo:` identical to `fl_encodeToFLEncoder:`.

 it would be simpler just to change it to `fl_encodeToFLEncoder:` in this source file, so there's only one method.